src/components/Calculator.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import calculate from '../logic/calculate';
+
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.calculation = this.calculation.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      total: null,
+      next: null,
+      operation: null,
+    });
+  }
+
+  shouldComponentUpdate() {
+    return true;
+  }
+
+  calculation(event) {
+    this.setState((obj) => calculate(obj, event.target.innerText));
+  }
+
+  render() {
+    const { next, operation, total } = this.state;
+    const display = next || operation || total || 0;
+
+    const numbers = [7, 8, 9, 4, 5, 6, 1, 2, 3, 0];
+    const operators = ['÷', 'x', '-', '+', '='];
+    const ids = {
+      7: 'seven',
+      8: 'eight',
+      9: 'nine',
+      4: 'four',
+      5: 'five',
+      6: 'six',
+      1: 'one',
+      2: 'two',
+      3: 'three',
+      0: 'zero',
+      '/': 'divide',
+      '*': 'multiply',
+      '-': 'subtract',
+      '+': 'add',
+    };
+
+    return (
+      <section className="calculator-container">
+        <div id="display" className="display">{ display }</div>
```

Comment:
- [ ] I would suggest you increase the height of your display so that it can accommodate failed modulus operation.

  ![vudej-display](https://user-images.githubusercontent.com/36542986/153192159-ae65cfed-d63a-45cb-9a23-e011df9d227f.jpg)